tests/unit/src/test/scala/tests/RenameLspSuite.scala

```diff
@@ -562,6 +562,50 @@ class RenameLspSuite extends BaseRenameLspSuite("rename") {
     fileRenames = Map.empty
   )
 
+  renamed(
+    "backtick-new-name",
+    """|/a/src/main/scala/a/Main.scala
+       |package a
+       |object Main{
+       |  val <<toRename>> = 123
+       |}
+       |/a/src/main/scala/a/Main2.scala
+       |package a
+       |object Main2{
+       |  val toRename = Main.<<toR@@ename>>
+       |}
+       |""".stripMargin,
+    newName = "other-rename"
+  )
+
+  renamed(
+    "backtick-old-and-new-name",
+    """|/a/src/main/scala/a/Main.scala
+       |package a
+       |object Main{
+       |  val <<`to-Rename`>> = 123
+       |}
+       |object Main2{
+       |  val toRename = Main.<<`to-R@@ename`>>
+       |}
+       |""".stripMargin,
+    newName = "`other-rename`"
+  )
+
+  renamed(
+    "backtick-not-needed",
```

Comment:
We will also now unbacktick anything that doesn't need backticks anymore. It will not backtick anything that already had them like when used in pattern matches:
```scala
val a = 12
val b = 12
a match{
  `b` => println("They are equal!")
}
```

Reply:
Actually, scratch that. We cannot unbacktick anything as it might still be needed in some cases like for example pattern matches.